uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -240,6 +241,67 @@ define([
           .then(function (response) {
             return response.values;
           });
+      },
+
+      /**
+       * Calls the getcomment backend API.
+       *
+       * @param {String} leaveRequestID - Id of leave request
+       * @param {Object} params
+       *
+       * @return {Promise}
+       */
+      getComments: function (leaveRequestID, params) {
+        params = _.assign(params, {
+          leave_request_id: leaveRequestID
+        });
+
+        return this.sendGET('LeaveRequest', 'getcomment', params)
+          .then(function (commentsData) {
+            return commentsData.values;
+          });
+      },
+
+      /**
+       * Calls the addcomment backend API.
+       *
+       * @param {String} leaveRequestID - ID of leave request
+       * @param {String} commentText - comment text
+       * @param {String} contactID - ID of contact
+       * @param {Object} params
+       *
+       * @return {Promise}
+       */
+      saveComment: function (leaveRequestID, commentText, contactID, params) {
+        params = _.assign(params, {
```

Comment:
if params is null then it will make it empty, better to start with empty object in assign
params = _.assign({}, params